File: NEW/TwoPointer/removeDuplicates.cpp
using namespace std;

#include <iostream>
#include <vector>
#include <unordered_map>

class RemoveDuplicates {
 public:
  static void remove(const vector<int>& arr) {
      int prev = arr[0];

      for(int it = 1; it<arr.size(); it++){
          if(arr[it] == prev){
              arr.erase(arr.begin());
          }
          prev=arr[it];
      }
  }
};

int main(int argc, char* argv[]) {
    vector<int> v = {2,3,3,3,6,9,9};
    RemoveDuplicates::remove(v); 
    for(int i=0; i<v.size(); i++)
        cout << v[i]<<", ";
    cout<< endl;
    
    v = {2,2,2,11};
    RemoveDuplicates::remove(v); 
    for(int i=0; i<v.size(); i++)
        cout << v[i]<<", ";
    cout<< endl;
}

File: Python/twoPointer/Remove_Duplicates.py
def func(arr):
    s, f = 0,1
    if(len(arr)==0):
        return 0
    while f < len(arr):
        if(arr[s]!=arr[f]):
            s+=1
            arr[s]=arr[f]
        f+=1

    return s+1


print(str(func([2, 3, 3, 3, 6, 9, 9])))
print(str(func([])))
print(str(func([1,2,3,4,5,6])))
print(str(func([1,1,1,1,1,1,1])))